creusot/tests/should_succeed/binary_search.stdout

```diff
@@ -246,48 +289,80 @@ module BinarySearch_Impl0_Len
+  clone BinarySearch_Get_Interface as Get3 with type t = uint32
+  clone BinarySearch_GetDefault_Interface as GetDefault2 with type t = uint32
+  clone BinarySearch_IsSorted_Interface as IsSorted0
+  val binary_search (arr : Type.binarysearch_list uint32) (elem : uint32) : Type.core_result_result usize usize
+    requires {IsSorted0.is_sorted arr}
+    requires {LenLogic1.len_logic arr <= Int32.to_int (1000000 : int32)}
+    ensures { forall x : (usize) . result = Type.Core_Result_Result_Err(x) -> (forall i : (int) . UInt64.to_int x < i && i < LenLogic1.len_logic arr -> elem < GetDefault2.get_default arr i (0 : uint32)) }
+    ensures { forall x : (usize) . result = Type.Core_Result_Result_Err(x) -> (forall i : (int) . Int32.to_int (0 : int32) <= i && i < UInt64.to_int x -> GetDefault2.get_default arr i (0 : uint32) < elem) }
+    ensures { forall x : (usize) . result = Type.Core_Result_Result_Ok(x) -> Get3.get arr (UInt64.to_int x) = Type.Core_Option_Option_Some(elem) }
+    
+end
 module BinarySearch_BinarySearch
   use mach.int.Int
   use mach.int.UInt32
   use mach.int.UInt64
   use prelude.Prelude
   use mach.int.Int32
   use Type
-  clone BinarySearch_LenLogic as LenLogic0 with type t = uint32
-  clone BinarySearch_GetDefault as GetDefault1 with type t = uint32
-  clone BinarySearch_Impl0_Len as Len2 with type t = uint32
-  clone CreusotContracts_Builtins_Resolve as Resolve3 with type self = Type.binarysearch_list uint32
-  clone CreusotContracts_Builtins_Resolve as Resolve4 with type self = uint32
-  clone CreusotContracts_Builtins_Resolve as Resolve5 with type self = bool
-  clone CreusotContracts_Builtins_Resolve as Resolve6 with type self = ()
-  clone CreusotContracts_Builtins_Resolve as Resolve7 with type self = usize
-  clone BinarySearch_Impl0_Index as Index8 with type t = uint32
   clone CreusotContracts_Builtins_Resolve as Resolve9 with type self = uint32
-  clone BinarySearch_IsSorted as IsSorted10
+  clone CreusotContracts_Builtins_Resolve as Resolve7 with type self = usize
+  clone CreusotContracts_Builtins_Resolve as Resolve6 with type self = ()
+  clone CreusotContracts_Builtins_Resolve as Resolve5 with type self = bool
+  clone CreusotContracts_Builtins_Resolve as Resolve4 with type self = uint32
+  clone CreusotContracts_Builtins_Resolve as Resolve3 with type self = Type.binarysearch_list uint32
   clone BinarySearch_Get as Get11 with type t = uint32
+  clone BinarySearch_IsSorted as IsSorted10 with function Get0.get = Get11.get
+  clone BinarySearch_GetDefault as GetDefault1 with type t = uint32, function Get0.get = Get11.get
+  clone BinarySearch_LenLogic as LenLogic0 with type t = uint32
+  clone BinarySearch_Impl0_Index_Interface as Index8 with type t = uint32,
+  function LenLogic0.len_logic = LenLogic0.len_logic, function Get1.get = Get11.get
+  clone BinarySearch_Impl0_Len_Interface as Len2 with type t = uint32,
+  function LenLogic0.len_logic = LenLogic0.len_logic
```

Comment:
Here is the key bit: we share the definition of `Get1` between `IsSorted10`, `GetDefault1` and `Index8`!